util/netcdf_generator.py

```diff
@@ -83,6 +83,37 @@ def _filter_params(self, ds, params):
                 ds = ds.drop(key)
         return ds
 
+    def _setup_coordinate_variables(self, ds):
+        """
+        Find the depth variable in the dataset and set assocciated coordinate attributes (10745)
+        :param ds: the dataset on which to operate
+        :return: the dataset with the coordinate variables and associated attributes setup
+        """
+        # find the depth_variable (10745 AC1)
+        depth_variable = find_depth_variable(ds.data_vars)
+        coordinates_key = 'coordinates'
+        coordinate_variables = "time lat lon"
+        if depth_variable:
+            coordinate_variables += " " + depth_variable
+        for var in ds.data_vars:
+            # coordinate variable shouldn't have coordinate attribute (10745 AC3)
+            # only scientific variables should have coordinate attribute (10745 AC2)
+            if var not in coordinate_variables \
+                and var not in app.config["NETCDF_NONSCI_VARIABLES"]:
```

Comment:
likely need to protect using `app.config.get('NETCDF_NONSCI_VARIABLES', [])`

Reply:
will do.

Reply:
done